src/components/Brushbox-textures.tsx

```diff
@@ -1,14 +1,15 @@
 import { useState } from "react";
 import './Brushbox-textures.css';
 import { Terrain } from "../Model/terrain";
+import { BrushSelection } from "../context";
 
-const BrushBoxTextures = ({ setBrushSelection }) => {
+const BrushBoxTextures = ({setBrushSelection} ) => {
     const [clicked, setClicked] = useState(-1)
     const [key, setKey] = useState(0)
 
-    const onClickIcon = (terrain: Terrain) => {
+    const onClickIcon = (terrain: Terrain) => { 
+        setBrushSelection({terrain} as BrushSelection)
```

Comment:
`{terrain}` should be a valid `BrushSelection`. I'm not sure if this cast is necessary, and actually somewhat unsafe. (see line 11 from the previous code to see that it works uncasted)